lib/standard/string_nit.c

```diff
@@ -11,10 +11,12 @@
 
 #include "string_nit.h"
 
+// Returns the length of `recv` as a `char*` (excluding the null character)
+long native_int_length_str(long recv){
+	return snprintf(NULL, 0, "%ld", recv);
```

Comment:
Ca pourrait marcher mais c'est fragile. snprintf est le mieux placé pour
mesurer la taille nécessaire du bluffer.
On pourrait imaginer des tas de saletés associés à la locale ou l'unicode
qui tomberaient comme ça d'un coup.

Une autre solution serait d'exposer `new NativeString` à la couche C qui
utiliserait ça au lieu du malloc pour allouer son buffer

On Wed, Feb 4, 2015 at 8:21 AM, Alexis Laferrière <notifications@github.com>
wrote:

> In lib/standard/string_nit.c
> <https://github.com/privat/nit/pull/1140#discussion_r24056200>:
>
> > @@ -11,10 +11,12 @@
> >
> >  #include "string_nit.h"
> >
> > +// Returns the length of `recv` as a `char*` (excluding the null character)
> > +long native_int_length_str(long recv){
> > +	return snprintf(NULL, 0, "%ld", recv);
>
> Have you considered doing this in Nit? It would probably require using
> floats, but it would be interesting to compare the perf... or look at how
> snprintf is implemented.
>
> var length = (self.pow10 + 1).floor
> if self < 0 then length += 1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/privat/nit/pull/1140/files#r24056200>.
>